nexus/reconfigurator/execution/src/lib.rs

```diff
@@ -184,5 +184,17 @@ where
     .await
     .map_err(|e| vec![anyhow!("{}", InlineErrorChain::new(&e))])?;
 
+    external_networking::ensure_zone_external_networking_deallocated(
```

Comment:
Mm I should've put my thoughts in a comment, although maybe we should have more formal documentation about the ordering here. For expunged zones I think it doesn't matter either way, because the planner refuses to reuse resources that haven't yet been removed. But for quiesced zones, my gut feeling is that it's important this deallocation happens _after_ we send the zones to sled agents? Otherwise sled-agent might still think it's still in use? This is kinda hand wavy since we can't actually quiesce zones today, so might be way off when we get there.

Reply:
It would definitely be good to capture this in a comment. I don't quite have a full understanding of the ordering constraints here, but I think for external DNS we may actually need to reuse the external IP within the same blueprint. In that case, we may need to deallocate IPs for expunged zones first.

Reply:
Hmm yeah fair point. I'll move it first and put a comment with both a note that it's ordered this way for IP reuse and that that might be wrong when we get to quiescing zones (e.g., if we deallocate an IP before we tell sled-agent to stop using it?).